lib/uklibparam/parser.c

```diff
@@ -664,16 +632,11 @@ static int parse_arg(struct parse_arg_ctx *ctx, char *strbuf, int scan_only)
 
 static inline void __reset_p_widx(void)
 {
-	struct uk_libparam_libdesc *ld;
 	struct uk_libparam_param *p;
-	__sz p_i;
 
-	UK_LIBPARAM_FOREACH_LIBDESC(ld) {
-		UK_LIBPARAM_FOREACH_PARAMIDX(ld, p_i) {
-			p = UK_LIBPARAM_PARAM_GET(ld, p_i);
-			UK_ASSERT(p);
-			p->__widx = 0;
-		}
+	UK_LIBPARAM_FOREACH_PARAM(p) {
+		UK_ASSERT(p);
+		p->__widx = 0;
```

Comment:
Yes, it is probably cleaner to move `__widx` to a different declaration so that we can keep the parameter struct `const`.